### PR TITLE
Tbolt/3646 subform updates

### LINKED ISSUE #3646: [Feature] Update FormAndReviewList to prepare for new sub-form changes
**Description and related issues -**

Add a Cancel button. Clicking Cancel should close the form and not save any thing that was changed in the form.

Change the Done button to Save.

Updated FormAndReviewList to not use auto-save. Clicking the Add X button should open a new form and that item should only be saved once the Save button is clicked.

### This task is done when…

This could include:

- [ ] a Cancel button is added
- [ ] Done button has been changed to Save
- [ ] nothing in the form is saved until the user clicks Save

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #3646 

**Description-**
Updates all subforms to the new design which includes:
* Adding a Save button to replace the Done button
* No longer auto-saves form inputs in subforms. Instead, clicking Save is the only way to save/update a subform item 
* Adding a cancel button which will close the subform without modifying it

**This pull request also touches…**

- One side effect of this change is that you can attempt to add multiple items but upon saving, only the corresponding item will be saved. This also goes for editing multiple items.

**Steps to manually verify this change...**

1. To verify the new behavior. Use all of the subforms and note the "Saving..." never appears until the Save button is clicked.

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/containers/activity/ContractorResource/ContractorResourceForm.test.js [hunk 1]
```diff
@@ -33,200 +33,184 @@ describe('the ContractorResourceForm component', () => {
...
-      .findWhere(
-        c =>
-          c.name() === 'ChoiceComponent' &&
-          c.prop('name') === 'apd-activity-contractor-hourly-key 1-no'
-      )
-      .simulate('change');
-    expect(props.setIsHourly).toHaveBeenCalledWith(43, 1, false);
-    expect(props.setCostForYear).toHaveBeenCalledWith(43, 1, '1066', '53792');
-  });
+//   test('renders if use hourly data is selected', () => {
```
Comment (thetif) [inline web/src/containers/activity/ContractorResource/ContractorResourceForm.test.js:48]: are we going to be using these tests later or can we delete them?

**Review by thetif — COMMENTED** [review]
code looks good

File: web/src/containers/activity/ContractorResource/ContractorResourceForm.test.js (hunk 1 above)
Comment (tbolt) [reply web/src/containers/activity/ContractorResource/ContractorResourceForm.test.js:48]: good catch, I thought I did delete them

**Review by thetif — APPROVED** [review]
Code looks good!

**Comment by thetif** [comment]
|Given|When|Then|Results|
|---|---|---|---|
|click add key personnel|user clicks cancel|nothing is added|:white_check_mark: PASS|
|click add key personnel|fill out form and click save|key personnel is added|:white_check_mark: PASS|
|click edit key personnel|make changes and click cancel|nothing is saved|:white_check_mark: PASS|
|click edit key personnel|make changes and click save|key personnel is updated|:white_check_mark: PASS|
|add 2 more key personnel|delete the first key personnel that can be deleted|the key personnel is removed|:white_check_mark: PASS|
|click add outcome|user clicks cancel|nothing is added|:white_check_mark: PASS|
|click add outcome|fill out form and click save|outcome is added|:white_check_mark: PASS|
|click edit outcome|make changes and click cancel|nothing is saved|:white_check_mark: PASS|
|click edit outcome|make changes and click save|outcome is updated|:white_check_mark: PASS|
|add another outcome|delete the first outcome|the outcome is removed|:white_check_mark: PASS|
|click add milestone|user clicks cancel|nothing is added|:white_check_mark: PASS|
|click add milestone|fill out form and click save|milestone is added|:white_check_mark: PASS|
|click edit milestone|make changes and click cancel|nothing is saved|:white_check_mark: PASS|
|click edit milestone|make changes and click save|milestone is updated|:white_check_mark: PASS|
|add another milestone|delete the first milestone|the milestone is removed|:white_check_mark: PASS|
|click add state staff|user clicks cancel|nothing is added|:white_check_mark: PASS|
|click add state staff|fill out form and click save|state staff is added|:white_check_mark: PASS|
|click edit state staff|make changes and click cancel|nothing is saved|:white_check_mark: PASS|
|click edit state staff|make changes and click save|state staff is updated|:white_check_mark: PASS|
|add another state staff|delete the first state staff|the state staff is removed|:white_check_mark: PASS|
|click add state expense|user clicks cancel|nothing is added|:white_check_mark: PASS|
|click add state expense|fill out form and click save|state expense is added|:white_check_mark: PASS|
|click edit state expense|make changes and click cancel|nothing is saved|:white_check_mark: PASS|
|click edit state expense|make changes and click save|state expense is updated|:white_check_mark: PASS|
|add another state expense|delete the first state expense|the state expense is removed|:white_check_mark: PASS|
|click add contractor|user clicks cancel|nothing is added|:white_check_mark: PASS|
|click add contractor|fill out form and click save|contractor is added|:white_check_mark: PASS|
|click edit contractor|make changes and click cancel|nothing is saved|:white_check_mark: PASS|
|click edit contractor|make changes and click save|contractor is updated|:white_check_mark: PASS|
|add another contractor|delete the first contractor|the contractor is removed|:white_check_mark: PASS|

**Comment by thetif** [comment]
@tbolt this looks good and I think it can move on to review, but I would like you to add more tests to the Basic APD tests. Can you add tests for clicking cancel on add and edit?

**Review by aricka-lewis — APPROVED** [review]
Looks good!

**Review by jeromeleecms — APPROVED** [review]
Works as advertised!

**Pull request merged**